neuralop/training/patching.py

```diff
@@ -78,125 +97,169 @@ def _stitch(self, x):
+        _, _, height, width = x.shape
+        padding = [
+            int(round(v))
+            for v in [
+                height * self.padding_fraction[0],
+                width * self.padding_fraction[1],
+            ]
+        ]
         self.padding_height = padding[0]
         self.padding_width = padding[1]
 
         patched = make_patches(x, n=2**self.levels, p=padding)
 
-        s1_patched = patched.size(-2) - 2*padding[0]
-        s2_patched = patched.size(-1) - 2*padding[1]
+        s1_patched = patched.size(-2) - 2 * padding[0]
+        s2_patched = patched.size(-1) - 2 * padding[1]
 
-        for level in range(1,levels+1):
+        for level in range(1, levels + 1):
             sub_sample = 2**level
-            s1_stride = s1_patched//sub_sample
-            s2_stride = s2_patched//sub_sample
-
-            x_sub = x[:,:,::sub_sample, ::sub_sample]
-
-            s2_pad = math.ceil((s2_patched + (2**levels - 1)*s2_stride - x_sub.size(-1))/2.0) + padding[1]
-            s1_pad = math.ceil((s1_patched + (2**levels - 1)*s1_stride - x_sub.size(-2))/2.0) + padding[0]
+            s1_stride = s1_patched // sub_sample
+            s2_stride = s2_patched // sub_sample
+
+            x_sub = x[:, :, ::sub_sample, ::sub_sample]
+
+            s2_pad = (
+                math.ceil(
+                    (s2_patched + (2**levels - 1) * s2_stride - x_sub.size(-1)) / 2.0
+                )
+                + padding[1]
+            )
+            s1_pad = (
+                math.ceil(
+                    (s1_patched + (2**levels - 1) * s1_stride - x_sub.size(-2)) / 2.0
+                )
+                + padding[0]
+            )
 
             if s2_pad > x_sub.size(-1):
                 diff = s2_pad - x_sub.size(-1)
-                x_sub = torch.nn.functional.pad(x_sub, pad=[x_sub.size(-1), x_sub.size(-1), 0, 0], mode='circular')
-                x_sub = torch.nn.functional.pad(x_sub, pad=[diff, diff, 0, 0], mode='circular')
+                x_sub = torch.nn.functional.pad(
+                    x_sub, pad=[x_sub.size(-1), x_sub.size(-1), 0, 0], mode="circular"
+                )
+                x_sub = torch.nn.functional.pad(
+                    x_sub, pad=[diff, diff, 0, 0], mode="circular"
+                )
             else:
-                x_sub = torch.nn.functional.pad(x_sub, pad=[s2_pad, s2_pad, 0, 0], mode='circular')
-            
+                x_sub = torch.nn.functional.pad(
+                    x_sub, pad=[s2_pad, s2_pad, 0, 0], mode="circular"
+                )
+
             if s1_pad > x_sub.size(-2):
                 diff = s1_pad - x_sub.size(-2)
-                x_sub = torch.nn.functional.pad(x_sub, pad=[0, 0, x_sub.size(-2), x_sub.size(-2)], mode='circular')
-                x_sub = torch.nn.functional.pad(x_sub, pad=[0, 0, diff, diff], mode='circular')
+                x_sub = torch.nn.functional.pad(
```

Comment:
That's another case where I find the formatted version actually marginally less readable